nautobot/docs/models/extras/dynamicgroup.md

```diff
@@ -1,7 +1,319 @@
+
+Continuing on to the children of `third-child`, the same iteration rules apply. The filter from `nested-child` gets applied to the filter being passed through from its parent, `third-child`, except that the "exclude" (boolean `NOT`) operator is still applied from the association of `third-child` to parent resulting in `(NOT nested-child`).
+
+!!! note
+    You'll see `NOT` emitted as an `AND NOT` because `NOT` is actually just shorthand for this. They are in fact interchangeable as Boolean operators. While it's technically possible to perform an `OR NOT` query from a database perspective, it is not supported by Dynamic Groups because the `NOT` operator maps directly to the "difference" set operation which is distinctly an `AND NOT` Boolean operation when performing underlying SQL queries at the database.
+
+Piecing the rules together, we ended up with two filters and by wrapping them in their own set of parentheses we get our final generated filter:
+
+```no-highlight
+((first-child OR second-child) AND (NOT nested-child))
+```
+
+How does this work?
+
+| Start with   | Filter        | Operator | Weight | Group Count | Final Count |
+| ------------ | ------------- | -------- | ------ | ----------- | ----------- |
+| parent       | none          | none     | none   | 389         | 389         |
+| first-child  | site=ams01    | restrict | 10     | 11          | 11          |
+| second-child | site=ang01    | include  | 20     | 7           | 18          |
+| third-child  | none          | none     | 30     | 389         | n/a         |
+| nested-child | status=active | exclude  | 10     | 389         | 0           |
+
+### Assigning Child Groups
+
+TBD. Higher-weight fields will be ordered lower within a form. (The default weight is 100.)
+
+## Dynamic Groups and the REST API
+
+Dynamic Groups are fully supported by the API.
+
+### Creating a Dynamic Group
+
+Request:
+
+```json
+POST /api/extras/dynamic-groups/
+{
+    "name": "parent2",
+    "slug": "parent2",
+    "description": "I am the parent group with nested children.",
+    "content-type": "dcim.device"
+}
+```
+
+Response:
+
+```json
+200 OK
+Content-Type: application/json
+
+{
+    "id": "03df0328-337c-4eeb-9095-fa2764cb31e2",
+    "display": "parent",
+    "url": "http://localhost:6789/api/extras/dynamic-groups/03df0328-337c-4eeb-9095-fa2764cb31e2/",
+    "name": "parent",
+    "slug": "parent",
+    "description": "I am the parent group with nested children.",
+    "content_type": "dcim.device",
+    "filter": {},
+    "children": [],
+    "created": "2022-07-06",
+    "last_updated": "2022-07-06T20:17:04.305663Z",
+    "custom_fields": {},
+    "computed_fields": {},
+    "relationships": {}
+}
+```
+
```

Comment:
In the REST API, can you assign a `filter` or `children` as part of an initial POST/`create` operation, or does it have to be a two-step process like the web UI requires?